wezterm-gui/src/tabbar.rs

```diff
@@ -373,6 +567,12 @@ impl TabBarState {
             line.insert_cell(x, black_cell.clone(), title_width, SEQ_ZERO);
         }
 
+        #[cfg(not(target_os = "macos"))]
```

Comment:
I'd love to eliminate as many of these `cfg` attributes from the bulk of the code as possible. Can we simply trust that the config is set appropriately by either the defaults or the user?

Reply:
I've found that titlebar buttons on macos behaves different than on windows and linux, for example when you hover buttons they react as single item, so I think it's better to use native macos buttons and somehow ignore decorations settings